### ISSUE TITLE
Add envsubst to CoreOS base install

### ISSUE THREAD

**Issue by rothgar** [coreos/bugs#1284]
During system provisioning there are often times I want to use systemd unit environment to update files/scripts during ExecStart. Without envsubst on the system this is very difficult to use the environment in scripts or config files. I have been trying hacky solutions with `eval` but it has problems because you have to use multiple commands just to get a file read by eval. I have been using something like `eval "echo \"$(</srv/manifests/addons/${i})\"" 2> /dev/null` but it has problems because quotes are stripped from the file. I also tried `sed` but there is a lot of overhead to read each environment variable and search for them in a loop and replace them in the config file.

I was using ignition to replace my metadata in files but I wanted to simplify my ignition configs (they were getting quite large) and make my systemd services not tied directly to when the system was provisioned.
I'm not sure the reasoning to add other (arguably) non-essental tools to CoreOS (git, vim, htop, etc.) but I would think envsubst could be one of the more simple and essential tools to have for configuration and deployments.

Just for clarification, envsubst is part of [GNU gettext tools](https://www.gnu.org/software/gettext/) and can read a file and output the text to stdout easily.

I'm open to discussion but I wanted to see if other people have better solutions or if there's a recommended way to replace environment variables in config files.

---

Other alternatives I've thought of but haven't explored yet.
- host json files in ignition and use metadata to compile the file. (would only work for json)
- statically compile envsubst and use it on the system
- use go template with another cli tool (does one exist already?)
- something like [dockerize](https://github.com/jwilder/dockerize) for config files

**Comment by jfoy** [comment]
Looks like this is addressed as of 1185.3.0 at least:
```
core@foo ~ $ cat /etc/os-release
NAME=CoreOS
ID=coreos
VERSION=1185.3.0
VERSION_ID=1185.3.0
BUILD_ID=2016-11-01-0605
PRETTY_NAME="CoreOS 1185.3.0 (MoreOS)"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
core@foo ~ $ which envsubst
/usr/bin/envsubst
core@foo ~ $ envsubst --version
envsubst (GNU gettext-runtime) 0.19.4
Copyright (C) 2003-2007 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Written by Bruno Haible.
```

**Comment by crawford** [comment]
Looks like `envsubst` was added back in 1053.2.0.
